CMake/SlicerMacroBuildLoadableModule.cmake

```diff
@@ -0,0 +1,326 @@
+    set_target_properties(${lib_name} PROPERTIES FOLDER ${LOADABLEMODULE_FOLDER})
+  endif()
+
+  set_property(GLOBAL APPEND PROPERTY SLICER_MODULE_TARGETS ${lib_name})
+
+  # --------------------------------------------------------------------------
+  # Install library
+  # --------------------------------------------------------------------------
+  if(NOT LOADABLEMODULE_NO_INSTALL)
+    install(TARGETS ${lib_name}
+      RUNTIME DESTINATION ${Slicer_INSTALL_QTLOADABLEMODULES_BIN_DIR} COMPONENT RuntimeLibraries
+      LIBRARY DESTINATION ${Slicer_INSTALL_QTLOADABLEMODULES_LIB_DIR} COMPONENT RuntimeLibraries
+      ARCHIVE DESTINATION ${Slicer_INSTALL_QTLOADABLEMODULES_LIB_DIR} COMPONENT Development
+      )
+  endif()
+
+  # --------------------------------------------------------------------------
+  # Install headers
+  # --------------------------------------------------------------------------
+  if(DEFINED Slicer_DEVELOPMENT_INSTALL)
+    if(NOT DEFINED ${LOADABLEMODULE_NAME}_DEVELOPMENT_INSTALL)
+      set(${LOADABLEMODULE_NAME}_DEVELOPMENT_INSTALL ${Slicer_DEVELOPMENT_INSTALL})
+    endif()
+  else()
+    if(NOT DEFINED ${LOADABLEMODULE_NAME}_DEVELOPMENT_INSTALL)
+      set(${LOADABLEMODULE_NAME}_DEVELOPMENT_INSTALL OFF)
+    endif()
+  endif()
+
+  if(NOT LOADABLEMODULE_NO_INSTALL AND ${LOADABLEMODULE_NAME}_DEVELOPMENT_INSTALL)
+    # Install headers
+    file(GLOB headers "${CMAKE_CURRENT_SOURCE_DIR}/*.h")
+    install(FILES
+      ${headers}
+      ${dynamicHeaders}
+      DESTINATION ${Slicer_INSTALL_QTLOADABLEMODULES_INCLUDE_DIR}/${LOADABLEMODULE_NAME} COMPONENT Development
+      )
+  endif()
+
+  if(BUILD_TESTING AND LOADABLEMODULE_WITH_GENERIC_TESTS)
+
+    set(KIT ${lib_name})
+
+    set(KIT_GENERIC_TEST_SRCS)
+    set(KIT_GENERIC_TEST_NAMES)
+    set(KIT_GENERIC_TEST_NAMES_CXX)
+    SlicerMacroConfigureGenericCxxModuleTests(${LOADABLEMODULE_NAME}
+      KIT_GENERIC_TEST_SRCS
+      KIT_GENERIC_TEST_NAMES
+      KIT_GENERIC_TEST_NAMES_CXX
+      )
+
+    set(CMAKE_TESTDRIVER_BEFORE_TESTMAIN "DEBUG_LEAKS_ENABLE_EXIT_ERROR();" )
+    create_test_sourcelist(Tests ${KIT}CxxTests.cxx
+      ${KIT_GENERIC_TEST_NAMES_CXX}
+      EXTRA_INCLUDE vtkMRMLDebugLeaksMacro.h
+      )
+
+    add_executable(${KIT}GenericCxxTests ${Tests})
+    set_target_properties(${KIT}GenericCxxTests PROPERTIES RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/${Slicer_BIN_DIR})
+    target_link_libraries(${KIT}GenericCxxTests ${KIT})
+    if(NOT "${LOADABLEMODULE_FOLDER}" STREQUAL "")
+      set_target_properties(${KIT}GenericCxxTests PROPERTIES FOLDER ${LOADABLEMODULE_FOLDER})
+    endif()
+
+    foreach(testname ${KIT_GENERIC_TEST_NAMES})
+      add_test(NAME ${testname}
+               COMMAND ${Slicer_LAUNCH_COMMAND} $<TARGET_FILE:${KIT}GenericCxxTests> ${testname})
+      set_property(TEST ${testname} PROPERTY LABELS ${KIT})
+    endforeach()
+  endif()
+
+endmacro()
+
+
+#
+# Deprecated macro - Will be removed for 4.4
+#
+macro(slicerMacroBuildQtModule)
```

Comment:
I would consider keeping this in the SlicerMacroBuildQtModule instead of moving it here. This way, when you do remove the macro, you have only one file to remove.